test/Makefile

```diff
@@ -18,21 +18,14 @@ clean:
 
 bins: $(BINS)
 
-bin/random: $(RANDOM_SRC)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/random $(RANDOM_SRC)/random
-
-bin/multihash: $(MULTIHASH_SRC)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/multihash $(MULTIHASH_SRC)/multihash
-
-bin/ipfs: $(IPFS_ROOT)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/ipfs $(IPFS_CMD)
-
-bin/pollEndpoint: $(POLLENDPOINT_SRC)/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/pollEndpoint $(POLLENDPOINT_SRC)
+bin/random: BIN_SOURCES = $(RANDOM_SRC)/random
+bin/multihash: BIN_SOURCES = $(MULTIHASH_SRC)/multihash
+bin/ipfs: BIN_SOURCES = $(IPFS_CMD)
+bin/pollEndpoint: BIN_SOURCES = $(POLLENDPOINT_SRC)
+
+bin/%: FORCE IPFS-BUILD-OPTIONS
+	@echo "*** building and installing test local version of '$*' binary ***"
```

Comment:
The goal was based on my subjective preference of not having to scan N instances of a logic (makefile recipe, function, etc) to see if there's any difference in them, when learning new code.

But I'm fine with abandoning this hunk of the PR.